auth.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const crypto = require('crypto');
+const DbInterface = require('./dbInterface.js');
+
+class Auth {
```

Comment:
I'll come back and review this later, this is a responsibility magnet that you'll veeeery likely need to split.
For now I'm focusing on architecture

Reply:
We'll destroy this monster soon!